examples/pcap_to_parquet.py

```diff
@@ -20,32 +20,51 @@ def ip_to_integer(s):
     return struct.unpack("!I", socket.inet_aton(s))[0]
 
 
+def get_ip_protocol(s):
+    if "tcp" in s:
+        return "tcp"
+    if "UDP" in s:
+        return "udp"
+    if "EIGRP" in s:
+        return "eigrp"
+    if "ICMP" in s:
+        return "icmp"
+    return None
+
+
 def to_parquet(filename, prefix="maccdc2012"):
     with open(filename) as f:
         traffic = {}
         nodes = set()
 
         for line in f.readlines():
+            if "unreachable" in line:
+                continue
             fields = line.split()
             if not fields:
                 continue
+            if fields[1] != "IP":
+                continue
+            protocol = get_ip_protocol(line)
+            if protocol not in ("tcp", "udp", "eigrp", "icmp"):
+                continue
             try:
                 addresses = []
 
                 # Extract source IP address and convert to integer
-                m = re.match(r'\d+\.\d+\.\d+\.\d+', fields[2])
+                m = re.match(r'(?P<address>\d+\.\d+\.\d+\.\d+)', fields[2])
```

Comment:
Good catch. Given the 8.5M rows for one hour worth of data, I assume compiling beforehand would definitely help.